templates/libvirt/xen.xml

```diff
@@ -25,7 +25,12 @@
         {% block os %}
             {% if vm.virt_mode == 'hvm' %}
                 <type arch="x86_64" machine="xenfv">hvm</type>
-                <loader>hvmloader</loader>
+                <!--
+                     For the libxl backend libvirt switches between OVMF (EFI)
+                     and SeaBIOS based on the loader type. This has nothing to
+                     do with the hvmloader binary.
+                -->
+                <loader type="{{ "rom" if vm.features.check_with_template('legacy-bios', True) else "pflash" }}">hvmloader</loader>
```

Comment:
While I'm looking at it, I think it would be more logical to have a feature with default value `False`. So, instead of `legacy-bios` (defaulting to `True`), better use `uefi` (defaulting to `False`).